app/models/user_response.rb

```diff
@@ -5,14 +5,17 @@ class UserResponse < ActiveRecord::Base
   belongs_to :user
   belongs_to :survey
   belongs_to :company
-  has_many :answers, inverse_of: :user_response, dependent: :destroy
+  has_many :answers, dependent: :destroy
 
   accepts_nested_attributes_for :answers
 
   after_create :create_response_activity
+  validate :check_email
```

Comment:
write descriptive method name 
check_email -> check_email_for guest_user or something like this